drivers/video/gc2145.c

```diff
@@ -0,0 +1,1145 @@
+			  struct video_format *fmt)
+{
+	struct gc2145_data *drv_data = dev->data;
+
+	*fmt = drv_data->fmt;
+
+	return 0;
+}
+
+static int gc2145_stream_start(const struct device *dev)
+{
+	const struct gc2145_config *cfg = dev->config;
+
+	return gc2145_write_reg(&cfg->i2c, 0xf2, 0x0f);
+}
+
+static int gc2145_stream_stop(const struct device *dev)
+{
+	const struct gc2145_config *cfg = dev->config;
+
+	return gc2145_write_reg(&cfg->i2c, 0xf2, 0x00);
+}
+
+static int gc2145_get_caps(const struct device *dev, enum video_endpoint_id ep,
+			   struct video_caps *caps)
+{
+	caps->format_caps = fmts;
+	return 0;
+}
+
+static int gc2145_set_ctrl(const struct device *dev, unsigned int cid, void *value)
+{
+	switch (cid) {
+	case VIDEO_CID_HFLIP:
+		return gc2145_set_ctrl_hmirror(dev, (int)value);
+	case VIDEO_CID_VFLIP:
+		return gc2145_set_ctrl_vflip(dev, (int)value);
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static const struct video_driver_api gc2145_driver_api = {
+	.set_format = gc2145_set_fmt,
+	.get_format = gc2145_get_fmt,
+	.get_caps = gc2145_get_caps,
+	.stream_start = gc2145_stream_start,
+	.stream_stop = gc2145_stream_stop,
+	.set_ctrl = gc2145_set_ctrl,
+};
+
+static int gc2145_init(const struct device *dev)
+{
+	struct video_format fmt;
+	int ret = 0;
+
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
+	const struct gc2145_config *cfg = dev->config;
+
+	ret = gpio_pin_configure_dt(&cfg->reset_gpio, GPIO_OUTPUT_ACTIVE);
+	if (ret) {
+		return ret;
+	}
+
+	gpio_pin_set_dt(&cfg->reset_gpio, 0);
+	k_sleep(K_MSEC(1));
+	gpio_pin_set_dt(&cfg->reset_gpio, 1);
+	k_sleep(K_MSEC(1));
+#endif
+
+	ret = gc2145_check_connection(dev);
+
+	if (ret) {
+		return ret;
+	}
+
+	gc2145_soft_reset(dev);
+	k_msleep(300);
+
+	gc2145_write_all(dev, default_regs, ARRAY_SIZE(default_regs));
+
+	/* set default/init format SVGA RGB565 */
+	fmt.pixelformat = VIDEO_PIX_FMT_RGB565;
+	fmt.width = UXGA_HSIZE;
+	fmt.height = UXGA_VSIZE;
+	fmt.pitch = UXGA_HSIZE * 2;
+	ret = gc2145_set_fmt(dev, VIDEO_EP_OUT, &fmt);
+	if (ret) {
+		LOG_ERR("Unable to configure default format");
+		return -EIO;
+	}
+	return ret;
+}
+
+/* Unique Instance */
+static const struct gc2145_config gc2145_cfg_0 = {
+	.i2c = I2C_DT_SPEC_INST_GET(0),
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
+	.reset_gpio = GPIO_DT_SPEC_INST_GET(0, reset_gpios),
+#endif
+};
+static struct gc2145_data gc2145_data_0;
+
+static int gc2145_init_0(const struct device *dev)
+{
+	const struct gc2145_config *cfg = dev->config;
+
+	if (!device_is_ready(cfg->i2c.bus)) {
+		LOG_ERR("Bus device is not ready");
+		return -ENODEV;
+	}
+
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
+	if (!gpio_is_ready_dt(&cfg->reset_gpio)) {
+		LOG_ERR("%s: device %s is not ready", dev->name, cfg->reset_gpio.port->name);
+		return -ENODEV;
+	}
+#endif
+
+	uint32_t i2c_cfg = I2C_MODE_CONTROLLER | I2C_SPEED_SET(I2C_SPEED_STANDARD);
+
+	if (i2c_configure(cfg->i2c.bus, i2c_cfg)) {
+		LOG_ERR("Failed to configure gc2145 i2c interface.");
+	}
```

Comment:
I don't really think this belongs to here, the speed is set from devicetree, the controller bit I'm not sure what it does, I realize this is taken from the ov2640 driver but it just does not look right, can you try and remove it and see if anything breaks? should probably go from there as well